src/libs/dotcms.api.js

```diff
@@ -2,10 +2,32 @@ const { login, isLogin, logout } = require('./dotcms.auth');
 const { getCurrentSite } = require('./dotcms.sites');
 const { esSearch } = require('./dotcms.esSearch');
 const fetch = require('node-fetch');
+let languagesConf;
+let language = {
+    code: '',
+    id: ''
+};
+
+const setLanguage = (pathname) => {
+    language.code = pathname.split('/')[1];
+    const langId = languagesConf.find((lang) => lang.languageCode === language.code);
+    if (langId) {
+        language.id = langId.id;
+    } else {
+        language.code = null;
+        language.id = 1;
+    }
+    return language;
+};
 
 const getUrl = (pathname) => {
+    if (language.code) {
+        let pathnameSplitted = pathname.split('/');
```

Comment:
All this logic to get the string of `en` or `es` is not necessary because it should be coming in the `react-router` I think in react router you can do: `path=":lang/*"` or something like that and get it from there, then with that string we can check if is in the lang listing and if not is english. Check the react router doc for https://reacttraining.com/react-router/web/example/ambiguous-matches